packages/stream_chat/lib/src/client/client.dart

```diff
@@ -289,15 +324,15 @@ class StreamChatClient {
     } catch (e, stk) {
       if (e is StreamWebSocketError && e.isRetriable) {
         final event = await _chatPersistenceClient?.getConnectionInfo();
-        if (event != null) return event;
+        if (event != null) return event.me;
```

Comment:
we should return `ownUser` in this too instead of `event.me` as the user properties might have changed over time.

We can do something like:
```dart
return event.me.merge(ownUser);
```